src/app/components/client/PageLoadEvent.tsx

```diff
@@ -18,25 +19,61 @@ export type Props = {
 // Empty component that records a page view on first load.
 export const PageLoadEvent = (props: Props) => {
   const [cookies, setCookie] = useCookies(["userId"]);
-  const userId = props.userId;
+  const pathname = usePathname();
 
   const recordTelemetry = useTelemetry();
 
-  const pageViewParams: GleanMetricMap["page"]["view"] = useMemo(() => {
+  const userId: GleanMetricMap["page"]["view"]["user_id"] = useMemo(() => {
     // If the user is not logged in, use randomly-generated user ID and store in cookie.
-    if (userId.startsWith("guest")) {
+    if (props.userId.startsWith("guest")) {
       if (!cookies.userId) {
-        setCookie("userId", userId);
+        setCookie("userId", props.userId);
       }
-      return { user_id: userId };
+      return props.userId;
     }
 
     if (props.enabledFlags.includes("FxaUidTelemetry")) {
-      return { user_id: userId };
+      return props.userId;
     } else {
-      return {};
+      return undefined;
     }
-  }, [cookies.userId, setCookie, userId, props.enabledFlags]);
+  }, [cookies.userId, setCookie, props.userId, props.enabledFlags]);
+
+  // On first load of the page, record a page view.
+  useEffect(() => {
+    const pageViewParams: GleanMetricMap["page"]["view"] = {
+      ...getUtmParams(),
+      path: pathname,
+    };
+    if (typeof userId === "string") {
+      pageViewParams.user_id = userId;
+    }
+
+    recordTelemetry("page", "view", pageViewParams);
+  }, [recordTelemetry, pathname, userId]);
+
+  // record attributions on page load
+  if (window.location.search?.length > 0) {
+    if (!localStorage.getItem("attributionsFirstTouch")) {
+      localStorage.setItem("attributionsFirstTouch", window.location.search);
+    }
+    localStorage.setItem("attributionsLastTouch", window.location.search);
+  }
```

Comment:
Does this need to be in a `try..catch` block?
I thinks `localStorage` can potentially throw a `SecurityError` if local storage is disabled. [1]
And looks like a `QuotaExceededError` if the target disk is full or the user declined a request for more space; when calling `setItem()`. [2]

[1] https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage#exceptions
[2] https://developer.mozilla.org/en-US/docs/Web/API/Storage/setItem#exceptions